### ISSUE TITLE
Porting Status

### ISSUE THREAD

**Issue by joelselvaraj** [ubports-beryllium/projectmanagement#20]
We will track the status of all main functions here instead of separate issues. 

- [x] Camera
    - [x] Front Camera Photo
    - [x] Back Camera Photo
    - [x] Front Camera Video Recording
    - [x] Back Camera Video Recording
    - [x] Apparmor issues. Sometimes Camera does not ask for Mic permissions and leads to video recording and playback errors
    - [x] Flashlight during photo

> 

- [x] Display
    - [x] Manual brightness control
    - [ ] Automatic brightness control ~(Workaround [here](https://forum.xda-developers.com/project-treble/trebleenabled-device-development/overlay-enable-night-light-adaptive-t3741965) )~
    - [ ] Auto off display during call when phone near ears. Needs proximity sensor working. On some boots, it works and some boots it doesnt.
    - [x] Automatic display timeout when idle

> 

- [x] Time and Date
    - [x] Set timezone (Halium-9 issue) (Workaround in #12 )
    - [x] Set time and date manually

> 

- [x] Fingerprint

> 

- [ ] Mouse and Touchpad support (Not tested yet)

> 

- [x] Audio
    - [x] Speaker working for media audio and ringtone
    - [x] Heaphones working for audio and calls 

> 

- [x] Battery & Charging
    - [x] Should be able to charge battery
    - [x] Charging symbol activated while charging. ~(Sometimes doesnt show as charging)~ Fixed in https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/kernel-xiaomi-beryllium/-/commit/c9f7b00f8bf0d2e65939217922a02f4a0a80a37e
    - [x] Fast Charging
    - [x] Better battery backup and screen-on time

> 

- [x] Touch Screen

> 

- [x] VPN Support (Should work. Follow this [Guide](https://ubports.com/blog/ubports-blog-1/post/using-vpn-in-ubuntu-touch-132). Comment below if it doesnt or create a new issue.

> 

- [x] Bluetooth
    - [x] Load Bluetooth driver. 
    - [x] Audio over bluetooth
    - [ ] Reliability (Sometimes doesnt pair properly or requires to initiate pairing from the other device)
    - [ ] Bluetooth Auto connect (Doubt if Ubports support this)
    - [ ] Automatic Audio routing during playback when bluetooth connected (Doubt if even Ubports support this)

> 

- [x] Cellular Data Support (Halium-9 Issue)
    - [x] Connect to cellular data service
    - [ ] Connect to cellular data service while roaming. (Not tested)
    - [x] Stock network speed
    - [x] ~When having dual sim, not able to switch data to sim2. works in sim1 only~ with latest CI builds, mobile data will work fine in both sims

> 

- [x] Cellular Call Support
    - [x] Connect to cellular network 
    - [x] 2-way call audio - 
    - [x] Multi party call conference
    - [ ] VoLTE - Ubports doesnt support them. 
    - [x] Dual sim call

> 

- [x] Anbox Support to run android apps. (Enabled in kernel. ~However the internet inside android apps doesnt work in the image provided by ubports. Needs to be rebuild with some modifications~) Fixed with a temporary workaround in https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/commit/b0cc9b92485128bd649fb5f0dc090b0e617eef08

**Comment by joelselvaraj** [comment]
General discussions over porting status and clarifications can be asked here.

**Comment by biswatmak** [comment]
Will it have snapd?

**Comment by joelselvaraj** [comment]
> 
> 
> Will it have snapd?

No. As of now there is not support for Snap in Ubports. We do only the porting here. The features are being worked by the Ubports Team though. You can read some discussions over here:
https://forums.ubports.com/topic/2744/snap-and-flatpak-apps
https://forums.ubports.com/topic/4446/snapd-for-snaps-how-to-also-flatpak-or-appimage

**Comment by DevMashru** [comment]
>     * [x]  Battery & Charging
>       * [ ]  Charging symbol activated while charging. (Sometimes doesnt show as charging)

Hey,
Just wanted to know if you had the commits(dated 29/06/2020) in your system/core. If not, it could potentially help you fix this bug.
https://github.com/pixeldust-project-caf/android_system_core/commits/quiche/healthd

**Comment by AMRUT118** [comment]
when we can expect stable build of this rom sir can u reply plze

**Comment by joelselvaraj** [comment]
> when we can expect stable build of this rom sir can u reply plze
> 

Hey, i will work on this during November. Got caught up with exams.

**Comment by branja6** [comment]
Hello, kind sir. I hope you will have enough time and be willing to port Ubuntu Touch completely in order to have a stable release on Poco F1. I'm keeping my fingers crossed.

**Comment by Nandanrmenon** [comment]
facing an issue in receiving a call, when one person calls the call ui shows two person and no audio goes through. also no data working. cant change the from 2g to lte
carrier : airtel

**Comment by Nandanrmenon** [comment]
> facing an issue in receiving a call, when one person calls the call ui shows two person and no audio goes through. also no data working. cant change the from 2g to lte
> carrier : airtel

also phone's not charging

**Comment by joelselvaraj** [comment]
> facing an issue in receiving a call, when one person calls the call ui shows two person and no audio goes through. also no data working. cant change the from 2g to lte
> carrier : airtel

Hey are sure you are not using dual sim? Currently a single sim on slot 2(important) is stable. Other combinations seems to cause these and various other issues.

**Comment by joelselvaraj** [comment]
Here are some updates. 

- I have fixed the battery discharging issue. 

- But sadly dual sim is still a problem. 

- I tried getting official support with OTA updates, but mobile data is not working in those builds. Its not upto Erfan's Ubuntu Touch GSIs yet. Once its stable, i can release it. The ubuntu touch recovery is already ported. So we can get proper installer support, once I/someone figure out why mobile data is not working in those builds.

So currently only battery drain issue is fixed. I think i should release this fix first. Rest can wait for on another update

**Comment by branja6** [comment]
That is amazing! Thank you very much, Joel.

p.s. I did face issues while using the alpha build. I couldn't use my micro sd card. I've tried with standard Android formatting (exfat) as well as formating it to ext4 and the device just couldn't see it.

**Comment by AMRUT118** [comment]
Can i use it now as a daily driver

**Comment by Nandanrmenon** [comment]
> > facing an issue in receiving a call, when one person calls the call ui shows two person and no audio goes through. also no data working. cant change the from 2g to lte
> > carrier : airtel
> 
> Hey are sure you are not using dual sim? Currently a single sim on slot 2(important) is stable. Other combinations seems to cause these and various other issues.

no i am not using dual sim and since you mentioned sim channel i was using it on slot 1. also i found an issue with wifi. what happened is i was testing arround if i could try to get data work so i turned off wifi and after sometime tried turning it back on, but then i connects then asks of password after a minute

**Comment by Nandanrmenon** [comment]
> Can i use it now as a daily driver
> […](#)
> On 26-Nov-2020, at 7:54 PM, branja6 ***@***.***> wrote: ﻿ That is amazing! Thank you very much, Joel. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

umm, not yet you can use but its really filled with bugs for a daily user

**Comment by zenMaya** [comment]
>  once I/someone figure out why mobile data is not working in those builds.

@joelselvaraj, Are there any guesses how to debug this? Can you maybe compare the builds with diff? I would really like to help, but I'm no expert in Android.

**Comment by Nandanrmenon** [comment]
@joelselvaraj, update: i switched to 2nd slot and sms and call are working fine but still data is giving headache. it would be great if we could make android apps work too.

**Comment by joelselvaraj** [comment]
@Nandanrmenon Good to know. Mobile data used to work fine for me on slot 2 without any issues. I can make android apps work, but last time i checked, internel inside android apps wont work and to fix them, i need to build custom anbox android image. I m currently hooked up in my academic work. It will atleast take month or two before i work on this again. I ll try add android apps support in between somehow if possible.

**Comment by joelselvaraj** [comment]
> 
> 
> > once I/someone figure out why mobile data is not working in those builds.
> 
> @joelselvaraj, Are there any guesses how to debug this? Can you maybe compare the builds with diff? I would really like to help, but I'm no expert in Android.

I did try diffs and stuff with the help from NotKit from Ubports too. We couldnt really find out the issue. I worked days on it to solve it with no solutions/workaround. I ll look into again later in a month or two.

**Comment by Nandanrmenon** [comment]
> @Nandanrmenon Good to know. Mobile data used to work fine for me on slot 2 without any issues. I can make android apps work, but last time i checked, internel inside android apps wont work and to fix them, i need to build custom anbox android image. I m currently hooked up in my academic work. It will atleast take month or two before i work on this again. I ll try add android apps support in between somehow.

actually i found the solution, apparently switching apn helped
edit: ubports is now much more usable than last time, thanks to you ;)

**Comment by Nandanrmenon** [comment]
update: calls, sms and data pretty much work good. well the next bug i am not sure if its gsi problem or device related problem, i found a gesture bug where i can't do anything and have to reboot quite often.

regarding restart, the phone shutdowns and doesn't wake up unless you hold the power button and manually turn on.

i tried using headsets and the sound still comes from phone speakers 🤷‍♀️.

**Comment by joelselvaraj** [comment]
Can you elaborate more on the audio issue? Wired/Bluetooth headphones? How did u test it?

**Comment by Nandanrmenon** [comment]
> Can you elaborate more on the audio issue? Wired/Bluetooth headphones? How did u test it?

I tired to open YouTube with the default browser and when I connected the earphones(wired), the audio was coming out of the speaker.

**Comment by RusJJ** [comment]
What's about MultiROM?
Trying to install using a zip «gsi installer». Looks like it's okay but then Recovery just crashes. Im still on Lineage (second ROM).

**Comment by CedArctic** [comment]
Haven't managed to install it myself yet but from some videos it seems that the notch support in UBPorts is getting in the way of some things. Found this patch if it interests anyone: https://github.com/JamiKettunen/unity8-notch-hax

**Comment by Nandanrmenon** [comment]
> Haven't managed to install it myself yet but from some videos it seems that the notch support in UBPorts is getting in the way of some things. Found this patch if it interests anyone: https://github.com/JamiKettunen/unity8-notch-hax

this can actually fix the notch issue. maybe i will try porting for poco myself

Edit: can someone test notch fix for me, cant install ubports now thats why

**Comment by PeterGamma** [comment]
One major issue of Xiaomi devices: it is difficult to find out which device to choose:

https://forum.xda-developers.com/f/ubuntu-touch-ports-to-android.2098/

**Comment by joelselvaraj** [comment]
> Here are some updates.
> 
> * I have fixed the battery discharging issue.
> * But sadly dual sim is still a problem.
> * I tried getting official support with OTA updates, but mobile data is not working in those builds. Its not upto Erfan's Ubuntu Touch GSIs yet. Once its stable, i can release it. The ubuntu touch recovery is already ported. So we can get proper installer support, once I/someone figure out why mobile data is not working in those builds.
> 
> So currently only battery drain issue is fixed. I think i should release this fix first. Rest can wait for on another update

Update Guys. Got to know about a workaround for the mobile data not working in official ubports build! Its to related to this. https://github.com/ubports/ubuntu-touch/issues/1657. 

**Steps to fix mobile data issue in official builds:** 
* Turn off the mobile data.
* ssh/open terminal app and run
```
/usr/share/ofono/scripts/set-context-property 0 Protocol ip
```
* Turn on mobile data.

Its a one time fix. it seems to persist after reboots. so first time after installation, we might have to do this temporarily. they ll eventually fix though!

**For those who want to try out Official Ubports Build:**
* Download the boot and system image from here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/993644033/artifacts/browse/out/
* wipe everything
* make sure you have android 9 firmware and vendor (by installing lin-16)
* format data
* fastboot flash system system.img
* fastboot flash boot boot.img
* reboot. thats it!
* Oh. Dont forget the temporary fix mentioned above! Do let me know if it works!

**Comment by Nandanrmenon** [comment]
> > Here are some updates.
> > 
> > * I have fixed the battery discharging issue.
> > * But sadly dual sim is still a problem.
> > * I tried getting official support with OTA updates, but mobile data is not working in those builds. Its not upto Erfan's Ubuntu Touch GSIs yet. Once its stable, i can release it. The ubuntu touch recovery is already ported. So we can get proper installer support, once I/someone figure out why mobile data is not working in those builds.
> > 
> > So currently only battery drain issue is fixed. I think i should release this fix first. Rest can wait for on another update
> 
> Update Guys. Got to know about a workaround for the mobile data not working in official ubports build! Its to related to this. [ubports/ubuntu-touch#1657](https://github.com/ubports/ubuntu-touch/issues/1657).
> 
> **Steps to fix mobile data issue in official builds:**
> 
> * Turn off the mobile data.
> * ssh/open terminal app and run
> 
> ```
> /usr/share/ofono/scripts/set-context-property 0 Protocol ip
> ```
> 
> * Turn on mobile data.
> 
> Its a one time fix. it seems to persist after reboots. so first time after installation, we might have to do this temporarily. they ll eventually fix though!
> 
> **For those who want to try out Official Ubports Build:**
> 
> * Download the boot and system image from here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/993644033/artifacts/browse/out/
> * wipe everything
> * make sure you have android 9 firmware and vendor (by installing lin-16)
> * fastboot flash system system.img
> * fastboot flash boot boot.img
> * reboot. thats it!
> * Oh. Dont forget the temporary fix mentioned above! Do let me know if it works!

Will let you know tomorrow evening. Have to test notch fix also. :)

**Comment by joelselvaraj** [comment]
> Will let you know tomorrow evening. Have to test notch fix also. :)

@Nandanrmenon
Regarding the notch fix, i just sent a pull request: https://github.com/JamiKettunen/unity8-notch-hax/pull/2
Hope it gets merged soon.

To try manually you can 
* `wget -O apply-notch-hax.sh https://git.io/JtIpG`
* `chmod +x apply-notch-hax.sh`
* run the `./apply-notch-hax.sh`. it will complain device not supported. (.cache/unity8-notch-hax folder is created this time)
* Download the beryllium.diff from here: https://raw.githubusercontent.com/ubports-beryllium/unity8-notch-hax/main/patches/beryllium.diff 
* copy it to `.cache/unity8-notch-hax/beryllium.diff`
* run `./apply-notch-hax.sh` again

Let me know if it works too. thanks! After it gets merged, first 3 steps is enough.

**Comment by RusJJ** [comment]
How to install as a second ROM using MultiROM? It shows an error 255, as i remember.

**Comment by joelselvaraj** [comment]
> How to install as a second ROM using MultiROM? It shows an error 255, as i remember.

as far as i know, there is no multirom support for ubports yet. Poco F1 only has multirom support for android os and not ubports.

**Comment by Squiglygon** [comment]
> **For those who want to try out Official Ubports Build:**
> 
>     * Download the boot and system image from here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/993644033/artifacts/browse/out/
> 
>     * wipe everything
> 
>     * make sure you have android 9 firmware and vendor (by installing lin-16)
> 
>     * fastboot flash system system.img
> 
>     * fastboot flash boot boot.img
> 
>     * reboot. thats it!
> 
>     * Oh. Dont forget the temporary fix mentioned above! Do let me know if it works!

Does it mean that this is all I need to install UT on a F1 that has LO16?
Can I skip everything here? https://forums.ubports.com/topic/4611/xiaomi-poco-f1-beryllium

**Comment by joelselvaraj** [comment]
> Does it mean that this is all I need to install UT on a F1 that has LO16?
> Can I skip everything here? https://forums.ubports.com/topic/4611/xiaomi-poco-f1-beryllium

Yes. If u already like Los 16 installed. Just format data and fastboot flash system and boot image.

**Comment by milikhin** [comment]
> **For those who want to try out Official Ubports Build:**
> 
>     * Download the boot and system image from here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/993644033/artifacts/browse/out/
> 
>     * wipe everything
> 
>     * make sure you have android 9 firmware and vendor (by installing lin-16)
> 
>     * fastboot flash system system.img
> 
>     * fastboot flash boot boot.img
> 
>     * reboot. thats it!
> 
>     * Oh. Dont forget the temporary fix mentioned above! Do let me know if it works!

Thank you, official build works fine for me! Mobile data also works out of the box without the temporary fix.

**Comment by joelselvaraj** [comment]
> Thank you, official build works fine for me! Mobile data also works out of the box without  the temporary fix.

Hey nice to know! U can try out the ubports recovery too. It has ota update support. If u could confirm Ota updates through ubports recovery works, it would be really helpful. Thanks.

**To test Ubports Recovery and ota updates**
- Download https://github.com/ubports-beryllium/halium_bootable_recovery/releases/download/v1/ubports-recovery.img
- `fastboot flash recovery ubports-recovery.img`
- In ubuntu touch settings app, you can check for updates. Wait till u get an update. Try to install from there. It should automatically reboot into recovery and install the updates. If u check the settings app again, the update shouldn't be there. Meaning it got installed.

**Comment by milikhin** [comment]
> Hey nice to know! U can try out the ubports recovery too. It has ota update support.

Thank you! I confirm that OTA updates work!

**Comment by AMRUT118** [comment]
Can I use it as daily driver now

**Comment by Nandanrmenon** [comment]
> > Thank you, official build works fine for me! Mobile data also works out of the box without  the temporary fix.
> 
> Hey nice to know! U can try out the ubports recovery too. It has ota update support. If u could confirm Ota updates through ubports recovery works, it would be really helpful. Thanks.
> 
> **To test Ubports Recovery and ota updates**
> 
> * Download https://github.com/ubports-beryllium/halium_bootable_recovery/releases/download/v1/ubports-recovery.img
> * `fastboot flash recovery ubports-recovery.img`
> * In ubuntu touch settings app, you can check for updates. Wait till u get an update. Try to install from there. It should automatically reboot into recovery and install the updates. If u check the settings app again, the update shouldn't be there. Meaning it got installed.

recovery installed, looks like los fork though 😬 + thanks for the notch fix and increased storage 😁

**Comment by Nandanrmenon** [comment]
@joelselvaraj new bug, wifi doesnt connect in some boots, two to three shutdowns make it work and sometimes i have to disconnect. second, fingerprint menu doesnt open up. :/

edit (02/03) : fingerprint bug was fixed in ubports update. there is one bug that keeps bothering, i.e. when i try to open recent, the gesture doesnt work. not sure if it is Lomiri and device related, restart only fixes it right now

**Comment by Nandanrmenon** [comment]
> > Thank you, official build works fine for me! Mobile data also works out of the box without  the temporary fix.
> 
> Hey nice to know! U can try out the ubports recovery too. It has ota update support. If u could confirm Ota updates through ubports recovery works, it would be really helpful. Thanks.
> 
> **To test Ubports Recovery and ota updates**
> 
> * Download https://github.com/ubports-beryllium/halium_bootable_recovery/releases/download/v1/ubports-recovery.img
> * `fastboot flash recovery ubports-recovery.img`
> * In ubuntu touch settings app, you can check for updates. Wait till u get an update. Try to install from there. It should automatically reboot into recovery and install the updates. If u check the settings app again, the update shouldn't be there. Meaning it got installed.

thanks for the recovery, the updates works but the unity8 notch fix is gone have to patch again

**Comment by svenha** [comment]
I bought a used Pocophone. The description here and on ubports.com assumes that the device should run LineageOS 16. Currently, I see only LineageOS 17 on their main pages. Will version 17 work or do I have to find a version 16 somewhere?

**Comment by RusJJ** [comment]
> I bought a used Pocophone. The description here and on ubports.com assumes that the device should run LineageOS 16. Currently, I see only LineageOS 17 on their main pages. Will version 17 work or do I have to find a version 16 somewhere?

You need LOS16. Link in XDA Topic.

**Comment by svenha** [comment]
Thanks @RusJJ. I will try the version from 2020-03-25.

**Comment by Nandanrmenon** [comment]
> > Thank you, official build works fine for me! Mobile data also works out of the box without  the temporary fix.
> 
> Hey nice to know! U can try out the ubports recovery too. It has ota update support. If u could confirm Ota updates through ubports recovery works, it would be really helpful. Thanks.
> 
> **To test Ubports Recovery and ota updates**
> 
> * Download https://github.com/ubports-beryllium/halium_bootable_recovery/releases/download/v1/ubports-recovery.img
> * `fastboot flash recovery ubports-recovery.img`
> * In ubuntu touch settings app, you can check for updates. Wait till u get an update. Try to install from there. It should automatically reboot into recovery and install the updates. If u check the settings app again, the update shouldn't be there. Meaning it got installed.

there is an issue with going back to twrp. i couldnt flash back twrp. here is the log

```
target reported max download size of 536870912 bytes
sending 'recovery' (32092 KB)...
FAILED (remote: Requested download size is more than max allowed)
finished. total time: 0.000s
```

**Comment by Nandanrmenon** [comment]
edit: my phones dead. cant flash anything, doesnt detect fastboot on both windows and linux

**Comment by joelselvaraj** [comment]
> edit: my phones dead. cant flash anything, doesnt detect fastboot on both windows and linux

I m not sure what happened. People did report it was working fine and updates get installed properly. Let me remove the file for now. I ll look into it. I think i m facing similar issue too.

**Comment by joelselvaraj** [comment]
@Nandanrmenon I literally faced the same issue in Ubuntu 20.04 now. Then when dual boot into windows, windows wouldnt even recognize my device. Here is what i did. I used a different windows computer and external usb 2.0 hub. Then it recognized. And i m able to flash them now. Do you by chance tried it out first in linux too?

**Comment by joelselvaraj** [comment]
Most likely related to these USB 3.0 fastboot problem. https://forum.xda-developers.com/t/fastboot-problem-on-usb-3-0-ports.3859995/. It initially didnt work and i couldnt flash anything. Then I bought usb 2.0 hub and i was able to flash them. Then in between, last few months, usb 3.0 fastboot worked for me. Then suddenly today, its not working again. I used a different computer and usb 2.0 hub, now its working. No idea whats with usb 3.0, fastboot and Xiaomi. Hope u ll be able to fix yours soon too.

**Comment by joelselvaraj** [comment]
@Nandanrmenon I released the recovery again. I tested it. The recovery is working fine. Its something to do with the system + usb 3.0 configuration. These type of fastboot issues are becoming common nowadays and big headache. All i can suggest is try a different pc, laptop, old ones preferrably with proper usb 2.0 ports.  Else buy cheap USB 2.0 hubs that are known to solve these problems. These fastboot errors are so random. So long, it was working in my computer. Since, u said its not working, i tried it and it suddenly didnt work for me too. Switched to another laptop and used an external usb 2.0 hub, now its working again.

**Comment by Nandanrmenon** [comment]
@joelselvaraj thank you, it was definitely 2.0 problem. i tried using my laptop with usb 2.0 port and used windows 10 with mi flash tool to install drivers.

**Comment by svenha** [comment]
Can the new ubports-recovery be used for installing ubports for the first time (avoiding the lineageos 16 step)?

**Comment by Nandanrmenon** [comment]
> Can the new ubports-recovery be used for installing ubports for the first time (avoiding the lineageos 16 step)?

Lineage 16 is required right now or else system won't boot

**Comment by L3O1337** [comment]
@Nandanrmenon @joelselvaraj is internet for anbox apps working now ??

**Comment by joelselvaraj** [comment]
> @Nandanrmenon @joelselvaraj is internet for anbox apps working now ??

I did fix it in my latest update with a [temporary workaround](https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/commit/b0cc9b92485128bd649fb5f0dc090b0e617eef08). It should work, if you fetch the latest boot.img and system.img from CI over here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/1101031496/artifacts/browse/out/

**Comment by L3O1337** [comment]
> > @Nandanrmenon @joelselvaraj is internet for anbox apps working now ??
> 
> I did fix it in my latest update with a [temporary workaround](https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/commit/b0cc9b92485128bd649fb5f0dc090b0e617eef08). It should work, if you fetch the latest boot.img and system.img from CI over here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/1101031496/artifacts/browse/out/

Got lil confuse just confirm me.. i have already lineage 16 so just flash those both images and done right

**Comment by Nandanrmenon** [comment]
> > > @Nandanrmenon @joelselvaraj is internet for anbox apps working now ??
> > 
> > 
> > I did fix it in my latest update with a [temporary workaround](https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/commit/b0cc9b92485128bd649fb5f0dc090b0e617eef08). It should work, if you fetch the latest boot.img and system.img from CI over here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/1101031496/artifacts/browse/out/
> 
> Got lil confuse just confirm me.. i have already lineage 16 so just flash those both images and done right

hello, you have clean flash los 16 making sure have the latest pie firmware(not ten). if you are already using los it is best to format data in twrp and flash boot and system img

**Comment by Nandanrmenon** [comment]
> > @Nandanrmenon @joelselvaraj is internet for anbox apps working now ??
> 
> I did fix it in my latest update with a [temporary workaround](https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/commit/b0cc9b92485128bd649fb5f0dc090b0e617eef08). It should work, if you fetch the latest boot.img and system.img from CI over here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/1101031496/artifacts/browse/out/

sir any chance of ubports installer?

**Comment by joelselvaraj** [comment]
> sir any chance of ubports installer?

Proper Ubports Recovery is needed for ubports installer to work. while i did have uploaded the recovery with OTA support. it doesnt seem to work properly now. The partitions are not getting mounted for some reason. Without it, the ubports installer is not installing the system image + boot image properly in the final stage. 

I need to work on this for few days to debug it, but unfortunately, this is my main phone. So i ll find time in the weekend or sometime to fix it. hopefully.

**Comment by L3O1337** [comment]
> > @Nandanrmenon @joelselvaraj is internet for anbox apps working now ??
> 
> I did fix it in my latest update with a [temporary workaround](https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/commit/b0cc9b92485128bd649fb5f0dc090b0e617eef08). It should work, if you fetch the latest boot.img and system.img from CI over here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/1101031496/artifacts/browse/out/

This boot and system not booting stuck on boot logo

**Comment by L3O1337** [comment]
@joelselvaraj man volte is not supported ??

**Comment by joelselvaraj** [comment]
Not supported by Ubports themselves. Nothing we can do abt it. I ll probably add it in the checklist. But no one is working on it as far as i know. Its proprietary and no one completely know abt how it works.
________________________________

**Comment by Nandanrmenon** [comment]
@joelselvaraj can we try halium-10.0 instead of halium-9.0 because it can a chaos installing back and forth the firmware
please do let us know if the recovery is working again or not

**Comment by Nandanrmenon** [comment]
can anyone give an guide how to install anbox

**Comment by Vistaus** [comment]
I was using Ubuntu Touch on the Poco F1 through Halium which mostly worked fine. However, I couldn't install the latest OTA so I thought I'd give the official UBPorts image a try, based on the instructions you posted over on XDA. I followed all of the steps carefully, but it just won't boot. It keeps displaying the Pocophone logo and that's it. It does boot into UBPorts Recovery when I hold volume up, so that confirms it was installed fine. Fastboot doesn't recognize the device, so it's pretty much bricked now. How do I recover from this (and also get the official UBPorts image working)?

**Comment by joelselvaraj** [comment]
Are you sure you u installed lineage 16 1st. then "format data" (not wipe data) before flashing the images?
________________________________

**Comment by Vistaus** [comment]
Yes, that's what I did.

**Comment by joelselvaraj** [comment]
Hmmm. No idea what's wrong then.
________________________________

**Comment by Vistaus** [comment]
But how do I recover from this now that only the UBPorts Recovery is working and fastboot isn't?

**Comment by joelselvaraj** [comment]
Fastboot should work fine. We are not doing anything that could break it. As i mentioned in above somewhere try using a different system with usb 2.0 hub.
________________________________

**Comment by Vistaus** [comment]
I did, but it still doesn't recognize it and fastboot even exits after a minute of trying and reboots the device.

**Comment by joelselvaraj** [comment]
Mm weird. I have kind of faced similar issues. But it would get recognized in that <1 min. U can try MiFlash tool to clean install fastboot stock rom. During that flash, it wouldn't restart automatically i guess. But try it on ur own risk. I flash stuffs every day and yet have not faced any unrecoverable brick.
________________________________

**Comment by Vistaus** [comment]
Didn't work either. I guess it's dead, Jim… (or so to speak)

**Comment by Nandanrmenon** [comment]
@joelselvaraj any updates sir?

**Comment by Nandanrmenon** [comment]
> Didn't work either. I guess it's dead, Jim… (or so to speak)

is your device showing up in fastboot devices?

**Comment by Vistaus** [comment]
> > Didn't work either. I guess it's dead, Jim… (or so to speak)
> 
> is your device showing up in fastboot devices?

Nope :(

**Comment by shubhambaid** [comment]
The link to the GSI image is inaccessible.

**Comment by Nandanrmenon** [comment]
> The link to the GSI image is inaccessible.

https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/1101031496/artifacts/browse/out/

**Comment by joelselvaraj** [comment]
@Nandanrmenon From here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/

You can click the latest job with "flashable" as name. Then click browse and then download from there. Whenever i create new CI builds, old ones are getting automatically by gitlab after sometime. I think that is the reason old link is not accessible.

**Comment by joelselvaraj** [comment]
@shubhambaid read this instruction from the previous message to get the latest images. GSI images are no longer maintained. Follow the post here: https://forum.xda-developers.com/t/rom-ubports-gsi-ubuntu-touch-for-poco-f1-pie-based-alpha-2-unofficial-updated.4133513/page-4#post-84559681

But use the link from above to download the system and boot image

**Comment by Nandanrmenon** [comment]
> @Nandanrmenon From here: https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1/xiaomi-beryllium/-/jobs/
> 
> You can click the latest job with "flashable" as name. Then click browse and then download from there. Whenever i create new CI builds, old ones are getting automatically by gitlab after sometime. I think that is the reason old link is not accessible.

any progress with bug fixes for Wi-Fi sir?

**Comment by joelselvaraj** [comment]
> any progress with bug fixes for Wi-Fi sir?

@Nandanrmenon what wifi issues u have?

**Comment by Nandanrmenon** [comment]
> > any progress with bug fixes for Wi-Fi sir?
> 
> @Nandanrmenon what wifi issues u have?

wifi automatically disconnects after some time and either restart or removing wifi settings and reconnecting helped me to get back online

**Comment by Nandanrmenon** [comment]
@joelselvaraj there is one more bug which is really annoying, the touch screen. when phone is idle for a few minutes, using the recent and app launcher is nearly impossible. touch almost doesnt work too. again reboot or again keeping it idle fixes it.

**Comment by L3O1337** [comment]
@joelselvaraj I have one doubt if I install jiovoice which allow jio user to make call without volte will it work in anbox ?

**Comment by joelselvaraj** [comment]
> @joelselvaraj I have one doubt if I install jiovoice which allow jio user to make call without volte will it work in anbox ?

Nah. i dont think so. Anbox dont have access to modem at all as far as i know.

**Comment by joelselvaraj** [comment]
Ubports is taking efforts to bring in support for VoLTE. Hopefully they ll make it work.

**Comment by L3O1337** [comment]
Thanks for the info I will wait until then

**Comment by L3O1337** [comment]
Congo 🥳🥳🥳 @joelselvaraj

**Comment by AMRUT118** [comment]
Is notch problem fixed !! And its listed on the Ubuntu website 😁 can i wait or should i use as daily driver pls suggest me and what about upi apps in india is that supported 🙆

**Comment by 00due** [comment]
<del>I'm pretty interested in this rom, but I dont want to replace android. I found a dual boot method for poco f1 called MultiROM. I asked if this supports MultiROM, but I haven't got any answer. I searched it on the comments of the XDA thread, but I think no one has answered. I also searched the internet, I found that MultiROM for nexus 5 supports ubuntu touch, but it was 7 years ago. Also, if you don't know either, does it soft brick, hard brick or bootloop if it doesn't work? I don't care of soft brick or bootloop but I'm scared of hard bricking my phone</del>. Thanks for the hard work of porting this to poco f1!

Edit: I found the answer: you can use it only as the primary rom.

**Comment by branja6** [comment]
> ~I'm pretty interested in this rom, but I dont want to replace android. I found a dual boot method for poco f1 called MultiROM. I asked if this supports MultiROM, but I haven't got any answer. I searched it on the comments of the XDA thread, but I think no one has answered. I also searched the internet, I found that MultiROM for nexus 5 supports ubuntu touch, but it was 7 years ago. Also, if you don't know either, does it soft brick, hard brick or bootloop if it doesn't work? I don't care of soft brick or bootloop but I'm scared of hard bricking my phone~. Thanks for the hard work of porting this to poco f1!
> 
> Edit: I found the answer: you can use it only as the primary rom.

You can ask your question here. I'm pretty sure some people have successfully dual-booted it. Ubuntu Touch for Poco F1
https://t.me/ubportsforpocof1

**Comment by RusJJ** [comment]
Is 3D acceleration here?

**Comment by niaproxy** [comment]
Hi.
I tried to apply workaround for automatic brightness control from XDA on recently flashed phone but it failed:

> chcon: invalid context: ‘u:object_r:vendor_overlay_file:s0’: No such file or directory

Did i miss something?

**Comment by joelselvaraj** [comment]
> Hi. I tried to apply workaround for automatic brightness control from XDA on recently flashed phone but it failed:
> 
> > chcon: invalid context: ‘u:object_r:vendor_overlay_file:s0’: No such file or directory
> 
> Did i miss something?

I dont think that workaround will work on Ubports now. I should probably remove it.

**Comment by joelselvaraj** [comment]
For everyone's info, if you are not aware. The work has been upstream'ed to Ubports community repos over here. Development happens over there : https://gitlab.com/ubports/community-ports/android9/xiaomi-poco-f1

You can check the latest status over here: https://devices.ubuntu-touch.io/device/beryllium/

**Comment by PeterGamma** [comment]
Thanks, Joel. You did a great job with the port of Uborts to the Poco F1.
